realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt

```diff
@@ -43,56 +55,64 @@ data class ProjectMetaData(
 /**
  * This class implements the Transform API provided by the Android Gradle plugin.
  */
-class RealmTransformer(project: Project) : Transform() {
-    private val logger: Logger = LoggerFactory.getLogger("realm-logger")
+class RealmTransformer(project: Project,
+                       private val inputs: ListProperty<Directory>,
+                       private val allJars: ListProperty<RegularFile>,
+                       private val referencedInputs: ConfigurableFileCollection,
+                       private val output: RegularFileProperty) {
     private lateinit var metadata: ProjectMetaData
     private var analytics: RealmAnalytics? = null
 
     init {
```

Comment:
Same, running all of this logic directly in the constructor feels ...iffy

Reply:
Same, I moved just exposed transform as internal and call it on the constructed object instead of implicitly from the init-block.